weaveworks/weave-gitops
# Add an additional page status for automations that summarizes their canaries

Part of [2572](https://github.com/weaveworks/weave-gitops-enterprise/issues/2572) (Enterprise issue)


An additional `PageStatus` component has been added to the `AutomationDetail` component that displays a summarized condition created by looking at all of the automation's canaries. 
It displays one of four possible messages created by the `createCanaryCondition` function (tests in `ReconciledObjectsTable.test.tsx`):
- No canaries
- One or more canaries are in progress 
- All canaries have succeeded 
- One or more canaries failed

Assumptions made in implementation:
- Canaries will only have the `Promoted` condition in their conditions array.
- `Succeeded` and `Initialized` are the only indicators of success, `Failed` is the only one for failed, and all others should count as progressing.

This will not affect OSS as the new status only displays when the `AutomationDetail` component has custom tabs or actions passed down to it. No follow-up PR in enterprise is needed.

To test:
- run `export PUSH_PROGRESSIVE_DELIVERY_MANIFESTS_TO_GITOPS_DEV_REPO=1 ./tools/reboot.sh` in enterprise
- checkout this branch in OSS, then from the ui-cra directory in enterprise, run `make core-ui && make core-lib`
- check out the enterprise UI for the kustomization `progressive-delivery-canary`

Screenshot:
<img width="1154" alt="image" src="https://user-images.githubusercontent.com/65822698/234658363-b36e7c56-739e-4850-92bc-96f921bee658.png">
